topology/probes/vpp/vpp.go

```diff
@@ -103,268 +124,508 @@ func (p *Probe) getInterfaceVrfID(ch api.Channel, index uint32) int64 {
 				p.Ctx.Logger.Debugf("Delete interface %v idx %d", name, index)
 				p.eventDelInterface(node)
-				delete(p.interfaceMap, index)
 			}
+			p.Ctx.Graph.Unlock()
+			delete(p.interfaceMap, index)
 		}
+	}
+
+	return nil
+}
 
-	nextEvents:
-		time.Sleep(VPPPollingTime * time.Millisecond)
+func (p *Probe) getHostSocketFilename(socketFilename string, n *graph.Node) (string, error) {
+	vppNode := topology.GetOwner(p.Ctx.Graph, n)
+	if vppNode == nil {
+		return "", fmt.Errorf("No owner found for node %s", n.ID)
 	}
 
-	ch.Close()
+	rootNode := topology.GetOwner(p.Ctx.Graph, vppNode)
+	if rootNode == nil {
+		return "", fmt.Errorf("No root owner found for node %s", vppNode.ID)
+	}
+
+	containerNode := p.Ctx.Graph.LookupFirstChild(rootNode, graph.Metadata{"Type": "container"})
+	if containerNode == nil {
+		return "", fmt.Errorf("No container found in node %s", rootNode.ID)
+	}
+
+	if dockerMetadata, err := containerNode.GetField("Docker"); err == nil {
+		dockerMetadata := dockerMetadata.(*docker.Metadata)
+		for _, mount := range dockerMetadata.Mounts {
+			if filepath.HasPrefix(socketFilename, mount.Destination) {
+				if hostPath, err := filepath.Rel(mount.Destination, socketFilename); err == nil {
+					return filepath.Join(mount.Source, hostPath), nil
+				}
+			}
+		}
+	}
+
+	return "", nil
 }
 
-// Start VPP probe and get all interfaces
-func (p *Probe) Start() {
-	conn, err := govpp.Connect(p.shm)
+func (p *Probe) indexSocketFilename(socketFilename string, intfNode *graph.Node) {
+	socketFilename, err := p.getHostSocketFilename(socketFilename, intfNode)
+
 	if err != nil {
-		p.Ctx.Logger.Error("VPP connection error: ", err)
+		p.Ctx.Logger.Error(err)
 		return
 	}
-	p.conn = conn
 
-	ch, err := conn.NewAPIChannel()
+	if socketFilename == "" {
+		return
+	}
+
+	if isMaster, _ := intfNode.GetFieldBool("VPP.Master"); isMaster {
+		p.masterMemifs.Index(intfNode.ID, intfNode, map[string]interface{}{socketFilename: nil})
+	} else {
+		p.slaveMemifs.Index(intfNode.ID, intfNode, map[string]interface{}{socketFilename: nil})
+	}
+
+	p.Ctx.Logger.Debugf("Indexed %s", socketFilename)
+}
+
+// Register a seed with the specified mount namespace
+func (p *Probe) startSeed(pid int64, root *graph.Node) (*exec.Cmd, error) {
+	if config.GetString("agent.auth.api.backend") != "noauth" {
+		return nil, errors.New("Starting seeds is only supported with the 'noauth' authentication backend")
+	}
+
+	seed := os.Getenv("SKYDIVE_SEED_EXECUTABLE")
+	if seed == "" {
+		seed = os.Args[0]
+	}
+	nsPath := fmt.Sprintf("/proc/%d/ns", pid)
+	sa, _ := common.ServiceAddressFromString(config.GetString("agent.listen"))
+	command := exec.Command(seed, "seed",
+		"--root="+string(root.ID),
+		"--pid="+path.Join(nsPath, "pid"),
+		"--mount="+path.Join(nsPath, "mnt"),
+		"--agent="+fmt.Sprintf("%s:%d", sa.Addr, sa.Port),
+		fmt.Sprintf("--filter=G.V('%s').Descendants().Subgraph()", root.ID),
+		"vpp")
+	command.Env = append(command.Env, "SKYDIVE_LOGGING_LEVEL=debug")
```

Comment:
we should take the log level from the config file ?

Reply:
Right